### PR TITLE
Include bindings for masm x86 and other x64 assemblers

### PR DESCRIPTION
**Pull request by mrfearless**

### CODE REVIEW

**Comment by aquynh** [comment]
cool, do you have any applications for this binding?

one thing: can you remove the binaries (*.lib), and ask users (via README) to compile & put them there instead?

**Comment by mrfearless** [comment]
Im updating the PR to include the readme and a sample program. I came across an access violation with calling ks_asm, which i've mentioned in #319 in relation to the address parameter - should be uint32_t for x86 and uint64_t for x64 - i replaced it with size_t in my test recompile to check and seems to work.

**Comment by aquynh** [comment]
the reason i used uint64_t for address parameter is that allows you to assemble code even if you are in 32bit. imagine this scenario: on Linux 32bit, you want to assemble 64bit instructions. in that case, size_t is 32bit, but you still want 64bit address, hence this 64bit parameter.

i believe that there is a solution for your problem. let me know when you can fix that, i will merge this PR. thanks!

**Comment by aquynh** [comment]
another thing: why the examples are in zip format, but not in source form?

File: bindings/masm/keystone_x64.inc [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+; In that case, ks_asm() would eventually return with error KS_ERR_ASM_SYMBOL_MISSING.
+;
+; To register the resolver, pass its function address to ks_option(), using
+; option KS_OPT_SYM_RESOLVER. typedef bool *ks_sym_resolver)(const char* symbol, uint64_t* value
+;
+; Invoke ks_option, ks, KS_OPT_SYM_RESOLVER, Addr ks_sym_resolver
+;
+
+ks_sym_resolver                 PROTO :QWORD, :QWORD ; const char* symbol, uint64_t* value
+
```
Comment (aquynh) [inline bindings/masm/keystone_x64.inc]: can you remove these blank lines at the end of this file, and the other files?

Comment (mrfearless) [reply bindings/masm/keystone_x64.inc]: Yes, I have done that now.

File: bindings/masm/KSExample_x64/KSExample_x64.Asm [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+	Invoke ks_close, ks_engine
+
+    ; Show encoded bytes to user
+    Invoke SetWindowText, hTxtEncodedBytes, Addr szFinalOutput
+
+    ret
+
+DoEncode endp
+
+
```
Comment (aquynh) [inline bindings/masm/KSExample_x64/KSExample_x64.Asm]: can you remove all these blank lines (and in other files)?

Comment (mrfearless) [reply bindings/masm/KSExample_x64/KSExample_x64.Asm]: Ok think I got them now, I removed the blank lines at the end of the text files and converted tabs to spaces as well

**Comment by aquynh** [comment]
but now the binary file bindings/masm/KSExample_x64.zip is still there?

**Comment by aquynh** [comment]
there are still some binary files, can we remove them too?

for consistency, can you rename readme.md to README.md

**Comment by aquynh** [comment]
can you please add your name to https://github.com/keystone-engine/keystone/blob/master/bindings/README ?

**Comment by mrfearless** [comment]
Any update?

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**